src/Hl7.Fhir.Base/Serialization/FhirJsonPocoDeserializer.cs

```diff
@@ -713,6 +714,16 @@ FhirJsonPocoDeserializerState state
                     new(contents, ERR.STRING_ISNOTAN_INSTANT.With(ref reader, contents));
             }
 
+            static (object?, FhirJsonException?) readLong(ref Utf8JsonReader reader)
+            {
+                // convert string in json to a long.
+                var contents = reader.GetString()!;
+
+                return long.TryParse(contents, out var parsed) ?
+                    new(parsed, null) :
+                    new(contents, ERR.STRING_ISNOTAN_INSTANT.With(ref reader, contents));
```

Comment:
That seems to be a copy/paste error (STRING_ISNOTAN_INSTANT), we're talking about longs here.

Reply:
Changed it to `ERR.NUMBER_CANNOT_BE_PARSED`